dev_docs/contributing/kibana_http_api_guidelines.mdx

```diff
@@ -99,23 +99,229 @@ resource "elasticstack_elasticsearch_index_lifecycle" "my_ilm" {
 }
 ```
 
+### Implement complete CRUD operations
+
+For Terraform to properly manage resources, your API must implement a complete set of CRUD operations on each resource. This is essential for the Terraform lifecycle (create, read, update, delete) to work properly.
+
+#### Required endpoints for each resource
+
+For every resource type, implement these HTTP endpoints:
+
+```
+GET    /api/resource/{id}         # Read - retrieve an existing resource
+POST   /api/resource              # Create - create a new resource
+PUT    /api/resource/{id}         # Update - update an existing resource
+DELETE /api/resource/{id}         # Delete - remove an existing resource
+GET    /api/resource              # List - retrieve all resources (with pagination)
+```
+
+<DocCallOut title="Terraform import">
+  The "List" operation is critical for Terraform's data source and import functionality ([see the docs](https://developer.hashicorp.com/terraform/cli/import)).
```

Comment:
++ to calling "list" out. ATM, some APIs don't have a list operation, forcing workarounds.